susemanager-utils/susemanager-sls/modules/runners/mgrk8s.py

```diff
@@ -44,13 +47,16 @@ def get_all_containers(kubeconfig=None, context=None):
     pods = api.list_pod_for_all_namespaces(watch=False)
     output = dict(containers=[])
     for pod in pods.items:
-        for container in pod.status.container_statuses:
-            res_cont = dict()
-            res_cont['container_id'] = container.container_id
-            res_cont['image'] = container.image
-            res_cont['image_id'] = container.image_id
-            res_cont['pod_name'] = pod.metadata.name
-            res_cont['pod_namespace'] = pod.metadata.namespace
-            output['containers'].append(res_cont)
+        try:
+            for container in pod.status.container_statuses:
```

Comment:
IIUC `container_statuses` is empty? In this case maybe in addition to the `try...except` block, a check could be added to avoid the exception altogether.

Reply:
Yes, you are right